R/filter_date_range.R

```diff
@@ -0,0 +1,28 @@
+#' Return location within the specified date range
+#'
+#' @param x A dataframe containing collar data with column "dt_col" that can be coerced into class "Date"
+#' @param date_col A character vector that specifies the name of date column in the supplied dataframe.
+#' @param start A character, date, or POSIXct string that can be coerced into a date.  Used to return locations after the specified date.
+#' @param end A character, date, or POSIXct string that can be coerced into a date.  Used to return locations before the specified date.
+#'
+#' @return A dataframe with locations recorded between the start date and end date provided.
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#'   filter_date_range(collar_data, start = "2000-12-30", end = "2019-01-30")
+#' }
+
+filter_date_range <- function(x,
+                              dt_col = "dt",
+                              start = "1900-01-01",
+                              end = (Sys.Date() + 1)) {
+
+  assertthat::assert_that(dt_col %in% names(x), msg = paste(dt_col, "is not a column name in the data frame supplied"))
+
```

Comment:
I think the reliance on a standard format here might be a bit difficult for users. In some ways I think you are now hitting the wall associated with not having a defined class for collar objects. If we had that you would know that there would be a column `dt_col` and it would have class `Date`. That said, if you like the looseness/informal nature of tibbles then I would probably suggest you check that the dt_col can be coerced to Date early because this will often be a point of failure, I think.

Reply:
Done.